src/floats.jl

```diff
@@ -72,6 +72,7 @@ function typeparser(::AbstractConf{BigFloat}, source, pos, len, b, code, pl, opt
         readbytes!(source, str, vlen)
         fastseek!(source, _pos) # reset IO to earlier position
     end
+    str = Base.cconvert(Cstring, String(str isa Vector{UInt8} ? copy(str) : str))
```

Comment:
This needs https://github.com/JuliaLang/julia/pull/56519 to be released in Julia 1.11 for this to work. Right now, this will truncate `str` if it is a `Memory`.